File: data/readable_format/cpp/file363.cpp
// loop_fusion.cpp
#include <iostream>

void clawloop();

int main() {
    clawloop();
    return 0;
}

void clawloop() {
    int iend = 2, jend = 4, kend = 2;

    for (int i = 0; i <= iend; ++i) {
        if (i == 0) {
            std::cout << "First iteration of i " << i << std::endl;
        }
        for (int j = 0; j <= jend; ++j) {
            for (int k = 0; k <= kend; ++k) {
                std::cout << "First loop body:" << i << '/' << j << '/' << k << std::endl;
            }
        }

        for (int j = 2; j <= jend; ++j) {
            for (int k = 0; k <= kend; ++k) {
                std::cout << "Second loop body:" << i << '/' << j << '/' << k << std::endl;
            }
        }
    }
}

File: data/readable_format/fortran/file363.f90
! loop_fusion.f90
PROGRAM LOOP_FUSION
  CALL clawloop
END PROGRAM LOOP_FUSION

SUBROUTINE clawloop
  IMPLICIT NONE
  INTEGER :: i
  INTEGER :: j
  INTEGER :: k
  INTEGER :: iend = 2
  INTEGER :: jend = 4
  INTEGER :: kend = 2

  DO i=0,iend
    IF (i == 0) THEN
      PRINT*, 'First iteration of i',i
    END IF
    DO j=0,jend
      DO k=0,kend
        PRINT *, 'First loop body:',i,'/',j,'/',k
      END DO
    END DO

    DO j=2,jend
      DO k=0,kend
        PRINT *, 'Second loop body:',i,'/',j,'/',k
      END DO
    END DO
  END DO
END SUBROUTINE clawloop